lib/ex_doc/formatter/html.ex

```diff
@@ -53,29 +66,46 @@ defmodule ExDoc.Formatter.HTML do
     %{config | main: main || @main}
   end
 
+  defp generate_build(extras, generated_files, output) do
+    extras = for {file, _, _} <- extras, do: "#{file}.html"
+
+    build_files =
+      MapSet.new(generated_files ++ extras)
```

Comment:
> Theoretically there is no need for MapSet.

I used `MapSet` to avoid duplicate entries coming from the `generate_api_reference/5` and `generate_extras/6` (example: `api-reference.html` and `#{config.main.html}), those duplicate entries make the build process fail at the beginning when we apply `File.rm!`, but, since we're changing `File.rm!` with `File.rm` is not necessary anymore, also, I'll restore `generate_api_reference/5` to its original position. That way we avoid to produce duplicate entries.

>  I would also call Enum.map(generated_files ++ extras, &[&1, "\n"]) to ensure all entries have a new line (including the last one). Keep in mind you don't need to convert to a string. File.write! accepts a list of binaries too.

@josevalim Awesome! with every review, I learn something new :)